cni/cmd/istio-cni/main.go

```diff
@@ -41,6 +47,15 @@ func main() {
 	}()
 
 	// TODO: implement plugin version
-	skel.PluginMain(plugin.CmdAdd, plugin.CmdCheck, plugin.CmdDelete, version.All,
+	err := skel.PluginMainWithError(plugin.CmdAdd, plugin.CmdCheck, plugin.CmdDelete, version.All,
 		fmt.Sprintf("CNI plugin istio-cni %v", istioversion.Info.Version))
+	if err != nil {
+		if err := err.Print(); err != nil {
```

Comment:
Isn't this just printing to stdout? We should do `log.Errorf` so it gets passed through our logger which ends up in the istio-cni daemonset (_as well as_ kubelet logs)

Reply:
This needs to print the JSON error to stdout as part of the cni spec ([here](https://github.com/containernetworking/cni/blob/main/SPEC.md#errors)). Do you mean that in addition we should also log that to `log.Errorf`? I'll add that.

Reply:
Though - `cmdAdd` also always logs the error, so this might get spammy. We could remove it there (but it contains more context about the error being from cmd-add).

As an aside - I changed the error log on failing to write to stdout to `Errorf`.